locustempus/main/views.py

```diff
@@ -118,22 +139,50 @@ def form_valid(self, form) -> HttpResponse:
         return result
 
 
-class CourseDetailView(LoggedInCourseMixin, DetailView):
-    model = Course
+class CourseDetailView(LoggedInCourseMixin, View):
     template_name = 'main/course_detail.html'
+    http_method_names = ['get', 'post']
 
-    def get_context_data(self, **kwargs):
-        ctx = super().get_context_data(**kwargs)
-        course: Course = kwargs.get('object')
-        ctx['projects'] = Project.objects.filter(course=course)\
-            .order_by('title')
-        ctx['is_faculty'] = course.is_true_faculty(self.request.user)
-        ctx['page_type'] = 'course'
-        ctx['breadcrumb'] = {
-            'Workspaces': reverse('course-list-view'),
-            course.title: '',
+    def post(self, request, *args, **kwargs):
+        course = get_object_or_404(Course, pk=kwargs.get('pk'))
+
+        is_grid = not request.session.get('project_list_grid', False)
+        request.session['project_list_grid'] = is_grid
+
+        ctx = {
+            'course': course,
+            'projects': Project.objects.filter(course=course)
```

Comment:
A minor point, I'd likely pull the queryset fetch into a local variable. This is a little hard to read. And, often, you need to go back and tune those with `prefetch`/`select_related`

Reply:
Got it, fixed.